src/Agent.Worker/Release/RetryExecutor.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
+{
+    public class RetryExecutor
+    {
+        private const int DefaultMaximumRetryCount = 5;
+
+        private const int DefaultMillisecondsToSleepBetweenRetries = 1000;
+
+        public int MaximumRetryCount { get; set; }
+
+        public int MillisecondsToSleepBetweenRetries { get; set; }
+
+        public Func<Exception, bool> ShouldRetryAction { get; set; }
```

Comment:
No caller ever sets this property. It looks like untested dead code you're adding into the repo. Perhaps have some PR coming on the heels of this one that will consume the code? If not, I would highly urge deleting untested dead code. Otherwise someone may think it's safe to use it and it's an unnecessary readability impact.